SPECS/shared-mime-info/shared-mime-info.spec

```
@@ -0,0 +1,671 @@
+# Disable pkgconfig autodep
+%global __requires_exclude ^%{_bindir}/pkg-config$
+BuildRequires:  docbook-dtd-xml
+BuildRequires:  docbook-style-xsl
+BuildRequires:  gcc
+BuildRequires:  gettext
+BuildRequires:  git
+BuildRequires:  glib-devel
+BuildRequires:  itstool
+BuildRequires:  libxml2-devel
+BuildRequires:  marinerui-rpm-macros
+BuildRequires:  meson
+BuildRequires:  xmlto
+
+%description
+This is the freedesktop.org shared MIME info database.
+
+Many programs and desktops use the MIME system to represent the types of
+files. Frequently, it is necessary to work out the correct MIME type for
+a file. This is generally done by examining the file's name or contents,
+and looking up the correct MIME type in a database.
+
+%prep
+%autosetup -S git
+tar xjf %{SOURCE6}
+
+mv xdgmime-*/ xdgmime/
+
+%build
+cd ./xdgmime/
+make
+cd ..
+%meson -Dupdate-mimedb=false -Dxdgmime-path=./xdgmime/
+%meson_build
+
+%install
+%meson_install
+
+find %{buildroot}%{_datadir}/mime -type d \
+| sed -e "s|^$RPM_BUILD_ROOT|%%dir |" > %{name}.files
+find %{buildroot}%{_datadir}/mime -type f -not -path "*/packages/*" \
+| sed -e "s|^$RPM_BUILD_ROOT|%%ghost |" >> %{name}.files
+
+mkdir -p %{buildroot}/%{_datadir}/applications
+install -m 644 %{SOURCE1} %{buildroot}/%{_datadir}/applications/gnome-mimeapps.list
+cat %{SOURCE2} >> %{buildroot}/%{_datadir}/applications/gnome-mimeapps.list
+cat %{SOURCE3} >> %{buildroot}/%{_datadir}/applications/gnome-mimeapps.list
+cat %{SOURCE4} >> %{buildroot}/%{_datadir}/applications/gnome-mimeapps.list
+cat %{SOURCE5} >> %{buildroot}/%{_datadir}/applications/gnome-mimeapps.list
+
+# Support fallback/generic mimeapps.list (currently based on gnome-mimeapps.list), see
+# https://lists.fedoraproject.org/pipermail/devel/2015-July/212403.html
+# https://bugzilla.redhat.com/show_bug.cgi?id=1243049
+cp %{buildroot}%{_datadir}/applications/gnome-mimeapps.list \
+   %{buildroot}%{_datadir}/applications/mimeapps.list
+
+## remove bogus translation files
+## translations are already in the xml file installed
+rm -rf %{buildroot}%{_datadir}/locale/*
+
+%check
+%meson_test
+
+%post
+/bin/touch --no-create %{_datadir}/mime/packages &>/dev/null ||:
+
+%transfiletriggerin -- %{_datadir}/mime
+update-mime-database -n %{_datadir}/mime &> /dev/null ||:
+
+%transfiletriggerpostun -- %{_datadir}/mime
+update-mime-database -n %{_datadir}/mime &> /dev/null ||:
+
+%files -f %{name}.files
+%license COPYING
+%doc README.md NEWS HACKING.md data/shared-mime-info-spec.xml
+%{_bindir}/*
+%{_datadir}/mime/packages/*
+%{_datadir}/applications/mimeapps.list
+%{_datadir}/applications/gnome-mimeapps.list
+# better to co-own this dir than to pull in pkgconfig
+%dir %{_datadir}/pkgconfig
+%{_datadir}/pkgconfig/shared-mime-info.pc
+%{_mandir}/man*/*
+%{_datadir}/gettext/its/shared-mime-info.its
+%{_datadir}/gettext/its/shared-mime-info.loc
+
+%changelog
+* Wed Jul 21 2021 Vinicius Jarina <vinja@microsoft.com> - 2.0-4
+- Rebuilt for Mariner Core UI
```

Missing license verification and attribution of the spec's source.

---

Done

---

Still missing the source attribution (no need to rebuild the package):

```suggestion
- Rebuilt for Mariner Core UI
- Initial CBL-Mariner import from Fedora 33 (license: MIT).
```

---

Done